lib/meilisearch/rails/multi_search/result.rb

```diff
@@ -0,0 +1,84 @@
+module MeiliSearch
+  module Rails
+    class MultiSearchResult
+      attr_reader :metadata
+
+      def initialize(searches, raw_results)
+        @results = {}
+        @metadata = {}
+
+        searches.zip(raw_results).each do |(index_target, search_options), result|
+          index_target = search_options[:class_name].constantize if search_options[:class_name]
+
+          @results[index_target] = case index_target
+                                   when String, Symbol
+                                     result['hits']
+                                   else
+                                     load_results(index_target, result)
+                                   end
+
+          @metadata[index_target] = result.except('hits')
+        end
+      end
+
+      include Enumerable
+
+      def each_hit(&block)
+        @results.each do |_index_target, results|
+          results.each(&block)
+        end
+      end
+      alias each each_hit
```

Comment:
Made the assumption that iterating through the return value of a multi search would iterate through the hits themselves, rather than a hash `search => hits` which can be done with `each_result`